crates/red_knot_server/src/server/api.rs

```diff
@@ -82,12 +85,17 @@ fn background_request_task<'a, R: traits::BackgroundDocumentRequestHandler>(
     Ok(Task::background(schedule, move |session: &Session| {
         let url = R::document_url(&params).into_owned();
 
-        let Ok(path) = url_to_system_path(&url) else {
+        let Ok(path) = url_to_any_system_path(&url) else {
             return Box::new(|_, _| {});
         };
```

Comment:
I think my preferred solution would be for `Index` to directly store a reference to `VirtualFile` or `File`. But I think you had the valid concern that the `Index` is independent of the `Workspace` and closing the workspace would result in having `File` pointers that point to a collected database. Although we could consider this a bug because the LSP should send a close event for all files when closing a workspace?

The benefit of storing the `File` on the `DocumentController` is that we can remove the need to convert between URLs.

Reply:
I saw your comment below.